### PR TITLE
Improve CircuitBreaker documentation #21879

### PR DESCRIPTION
**Pull request by Chuwiey**
Added additional documentation to CircuitBreaker. Hopefully this addresses all issues raised in #21879 - Can add more as needed.

Side note: At the moment I couldn't find a link to the CircuitBreakerProxy actor anywhere from the main docs. I added it as a note in the initialization, but wanted to ask here whether it made sense to combine the two?

### CODE REVIEW

File: akka-docs/rst/common/circuitbreaker.rst [hunk 1]
```diff
@@ -53,50 +53,56 @@ What do they do?
 
 .. image:: ../images/circuit-breaker-states.png
 
-========
+
+===============
 Examples
-========
+===============
```
Comment (johanandren) [inline akka-docs/rst/common/circuitbreaker.rst]: Prtty sure this should be the same number of `=` characters as the header text

File: akka-docs/rst/common/circuitbreaker.rst [hunk 2]
```diff
@@ -53,50 +53,56 @@ What do they do?
...
 
 --------------
 Initialization
 --------------
 
-Here's how a :class:`CircuitBreaker` would be configured for:
+Following are examples for initializing a :class:`CircuitBreaker` pattern with the following configuration:
```
Comment (johanandren) [inline akka-docs/rst/common/circuitbreaker.rst]: I liked original text better here.

File: akka-docs/rst/common/circuitbreaker.rst [hunk 3]
```diff
@@ -53,50 +53,56 @@ What do they do?
...
   * 5 maximum failures
   * a call timeout of 10 seconds 
   * a reset timeout of 1 minute
 
-^^^^^^^
+.. note::
+	
+	There is also a :class:`CircuitBreakerProxy` actor that you can use, it is an alternative implementation of the pattern, where the main difference is that it is intended to be used only for request-reply interactions with another actor. See :ref:`Circuit Breaker Actor <circuit-breaker-proxy>`
```
Comment (johanandren) [inline akka-docs/rst/common/circuitbreaker.rst]: This is not the most important thing to note, I'd move it down after the samples, but before the low level api section.

minor: I think we usually line break these before 120 chars

**Review by johanandren — COMMENTED** [review]
Thanks! A few comments in there. Did you try rendering it so you saw that it looks correct/nice?

File: akka-docs/rst/common/circuitbreaker.rst (hunk 1 above)
Comment (Chuwiey) [inline akka-docs/rst/common/circuitbreaker.rst]: Definitely rendered it, looked good. I can edit the "===" to match

File: akka-docs/rst/common/circuitbreaker.rst (hunk 3 above)
Comment (Chuwiey) [inline akka-docs/rst/common/circuitbreaker.rst]: Can do.

**Comment by Chuwiey** [comment]
Updated PR according to comments.

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

**Comment by Chuwiey** [comment]
great, thx! So who does the actual merge once someone approves?

**Comment by ktoso** [comment]
Anyone from the team once it hits 2 LGTMs ;) Don't worry, will be in soon.

**Comment by ktoso** [comment]
The process is documented here https://github.com/akka/akka/blob/master/CONTRIBUTING.md

**Comment by Chuwiey** [comment]
@ktoso good, then i can go on to an actual coding commit next ;-)

**Comment by Chuwiey** [comment]
ping (just in case it was forgotten during the holidays)

**Comment by patriknw** [comment]
OK TO TEST

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
@Chuwiey This is looking good and can be merged. Plesase squash it to one commit?

**Comment by Chuwiey** [comment]
squashed :)

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

**Comment by patriknw** [comment]
Thanks, @Chuwiey

**Pull request merged**